docs/products/kafka/howto/connect-with-go.rst

```diff
@@ -0,0 +1,278 @@
+Variable                     Description
+========================     =======================================================================================================
+``HOST``                     Host name for the connection
+``SSL_PORT``                 Port number to use for SSL
+``SASL_PORT``                Port number to use for SASL
+``SASL_USERNAME``            Name of the user for the connection
+``SASL_PASSWORD``            Password required to connect using SASL
+``CLIENT_ID``                Application-specific client id
+========================     =======================================================================================================
+
+With library ``sarama``
+-------------------------
+Install the library `Sarama <https://github.com/Shopify/sarama>`_ and use code snippet according to your preferred authentication method below.
+
+With SSL authentication
+***********************
+
+Set up properties to connect to the cluster:
+
+.. code:: go
+
+    package main
+
+    import (
+        "crypto/tls"
+        "crypto/x509"
+        "io/ioutil"
+        "log"
+        "github.com/Shopify/sarama"
+    )
+
+    func main() {
+        keypair, err := tls.LoadX509KeyPair("service.cert", "service.key")
+        if err != nil {
+            log.Println(err)
+            return
+        }
+
+        caCert, err := ioutil.ReadFile("ca.pem")
+        if err != nil {
+            log.Println(err)
+            return
+        }
+        caCertPool := x509.NewCertPool()
+        caCertPool.AppendCertsFromPEM(caCert)
+
+        tlsConfig := &tls.Config{
+            Certificates: []tls.Certificate{keypair},
+            RootCAs: caCertPool,
+        }
+
+        // init config, enable errors and notifications
+        config := sarama.NewConfig()
+        config.Producer.Return.Successes = true
+        config.Net.TLS.Enable = true
+        config.Net.TLS.Config = tlsConfig
+        config.Version = sarama.V0_10_2_0
+
+        brokers := []string{"{HOST}:{SSL_PORT}"}
+
+        producer, err := sarama.NewSyncProducer(brokers, config)
+
+        // add your logic
+    }
+
+
+With SASL-SSL authentication
+*****************************
+
+Set up properties to connect to the cluster:
+
+.. code:: go
+
+    package main
+
+    import (
+        "crypto/tls"
+        "crypto/x509"
+        "github.com/Shopify/sarama"
+        "io/ioutil"
+    )
+
+    func main() {
+        caCert, err := ioutil.ReadFile("ca.pem")
+        if err != nil {
+            panic(err)
+        }
+        caCertPool := x509.NewCertPool()
+        caCertPool.AppendCertsFromPEM(caCert)
+
+        tlsConfig := &tls.Config{
+            RootCAs: caCertPool,
+        }
+
+        // parse Kafka cluster version
+        version, err := sarama.ParseKafkaVersion("2.4.0")
+        if err != nil {
+            panic(err)
+        }
```

Comment:
This part looks odd. Not really needed at all? (also see later assignment of `config.Version = version` which is not there non-sasl snippet.